File: docs/.validation/csharp/error-handling_21.cs
// Source: hooks/error-handling.md:142
var session = await client.CreateSessionAsync(new SessionConfig
{
    Hooks = new SessionHooks
    {
        OnErrorOccurred = (input, invocation) =>
        {
            Console.Error.WriteLine($"[{invocation.SessionId}] Error: {input.Error}");
            Console.Error.WriteLine($"  Context: {input.ErrorContext}");
            Console.Error.WriteLine($"  Recoverable: {input.Recoverable}");
            return Task.FromResult<ErrorOccurredHookOutput?>(null);
        },
    },
});

File: docs/.validation/python/error-handling_21.py
# Source: hooks/error-handling.md:106
import asyncio

async def main():
    async def on_error_occurred(input_data, invocation):
        print(f"[{invocation['session_id']}] Error: {input_data['error']}")
        print(f"  Context: {input_data['errorContext']}")
        print(f"  Recoverable: {input_data['recoverable']}")
        return None
    
    session = await client.create_session({
        "hooks": {"on_error_occurred": on_error_occurred}
    })

asyncio.run(main())